File: 383/js/383.js
var canConstruct = function(ransomNote, magazine) {
    let m = ransomNote.length;
    let n = magazine.length;
    let magazineArray = magazine.split('');
    for (let i = 0; i < m; i ++){
        let find = false;
        for (let j = 0; j < n; j++){
            if (ransomNote[i] === magazineArray[j]){
                magazineArray[j] = "-";
                find = true;
                break;
            }
        }
        if (find == false){
            return false;
        }
    }

    return true;
};

File: 383/go/383.go
func canConstruct(ransomNote string, magazine string) bool {
    checkMap := make(map[rune]int)

    for _, str := range magazine {
        checkMap[str] = checkMap[str] + 1
    }

    for _, str := range ransomNote {
        if checkMap[str] <= 0 {
            return false
        }

        checkMap[str] = checkMap[str]  - 1
    }

    return true
}